lib/srv/db/redis/connection.go

```diff
@@ -0,0 +1,109 @@
+/*
+
+ Copyright 2022 Gravitational, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+*/
+
+package redis
+
+import (
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// DefaultPort is the Redis default port.
+const DefaultPort = "6379"
+
+const (
+	URIScheme    = "redis"
+	URISchemeSSL = "rediss"
+)
+
+// ConnectionMode defines the mode in which Redis is configured. Currently, supported are single and cluster.
+type ConnectionMode string
+
+const (
+	// Single mode should be used when connecting to a single Redis instance.
+	Single ConnectionMode = "single"
```

Comment:
I was looking for a better name and I think that `standalone` sounds better. I'll update it later today.